### PR TITLE
Inventor robot exporter refactor and bug-fixes

### PR DESCRIPTION
**Pull request by wangl5**
### Inventor robot exporter refactor

- Organized class/folder structure
- Refactored and fixed bugs in logic for detecting when the exporter environment is opened/closed - - Added logic for detecting when the exporter environment is shown/hidden, which occurs when the user switches to a document in Inventor which doesn't have the exporter loaded. This allows us to fix issues where docked panes wouldn't re-show after being hidden in other documents.
- Moved logic for detecting environment open/close/show/hide into a separate class, `SingleEnvironmentAddInServer`, which `RobotExporterAddInServer` inherits from
- New `HighlightManager` class that deals with joint highlighting and DOF highlighting, to simplify highlight logic and make the highlighting logic more maintainable
- Added a bunch of utility classes to separate out commonly used static helper functions
- Moved nearly all environment button handler logic out of `RobotExporterAddInServer` and into helper classes
- Replace button handler methods with lambdas
- Separate dockable window logic into separate classes (DOFKey, Advanced joint editor, Guide)
- Fixed issues with advanced joint editor highlight not being cleared after closing using the `DockableWindows.Events.OnHide` event
- Remove all singleton fields in classes except for RobotExporterAddInServer
- Refactor and remove a lot of code that was only needed for the standalone robot exporter
- Removed deprecated standalone GUI class, robot data loading/saving is now dealt with in the new `RobotDataManager` class
- Created separate classes for main mesh exporting and skeleton building methods
- Added async `ExportMeshes` and `BuildSkeleton` methods to the `BuildingSkeletonForm` and `ExportingMeshesForm` to make the `LoadMeshes` method in `RobotDataManager` more readable
- Added separate AddInSettings class for add-in specific settings

|    DOD    | Done |    Reviewer    | Issue | Notes |
|-----------|------|----------------|-------|-------|
| Code      | ❌   | Nicolas Espinoza/Alex Carter|  1090|       |
| Unit Test | N/A  | N/A            |       |       |
| Merged    | ❌  |                |       |       |

### CODE REVIEW

**Comment by KiraCorbett** [comment]
Was this going to be removed with the new default plane or did we decide on keeping it?image

**Comment by wangl5** [comment]
@KiraCorbett Yes, I was originally planning to remove the "start with field" option in this PR but since it was more of a change than a bug fix, I put it in #387 instead.

**Comment by matthew17754** [comment]
@KiraCorbett I am unable to produce your issue.

@wangl5 Everything looks good, but I'd recommend looking into the suggestions by @nicolasespinoza before we merge

File: exporters/BxDRobotExporter/BxDRobotExporter/GUI/Loading/BuildingSkeletonForm.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                MessageBox.Show(e.Error.Message);
+            }
+            else
+            {
+                DialogResult = DialogResult.OK;
+                Close();
+            }
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (wangl5) [reply exporters/BxDRobotExporter/BxDRobotExporter/GUI/Loading/BuildingSkeletonForm.cs]: Good idea! It should display for the full duration in #388

File: exporters/BxDRobotExporter/BxDRobotExporter/GUI/Guide/Guide.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                embeddedGuidePane = null;
+            }
+        }
+
+        public void TemporaryHide()
+        {
+            SoftSetVisibility(false);
+        }
+    }
+}
```
[the remark this answers is not in the document]
Comment (wangl5) [reply exporters/BxDRobotExporter/BxDRobotExporter/GUI/Guide/Guide.cs]: Good point! Addressed this in 1f655a1327875cfc2c54e4831a088bf77280bc58

File: exporters/BxDRobotExporter/BxDRobotExporter/SingleEnvironmentAddInServer.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+using BxDRobotExporter.Messages;
+using BxDRobotExporter.Properties;
+using BxDRobotExporter.Utilities;
+using Inventor;
+using Application = Inventor.Application;
+using Environment = Inventor.Environment;
+
+namespace BxDRobotExporter
+{
+    public abstract class SingleEnvironmentAddInServer : ApplicationAddInServer
```
[the remark this answers is not in the document]
Comment (wangl5) [reply exporters/BxDRobotExporter/BxDRobotExporter/SingleEnvironmentAddInServer.cs]: Added class description in c5d52af38484f687a495177e0e20b2fa4d22982d

File: exporters/BxDRobotExporter/BxDRobotExporter/Utilities/WinFormsUtils.cs [hunk 4]
```diff
@@ -60,5 +60,19 @@ public static void AddControlToNewTableRow(Control control, TableLayoutPanel tab
                     break;
             }
         }
+
+        public static void RecursiveControlNavigator(Control control, Action<Control> action)
```
[the remark this answers is not in the document]
Comment (wangl5) [reply exporters/BxDRobotExporter/BxDRobotExporter/Utilities/WinFormsUtils.cs]: If I recall correctly, the mouse events didn't seem to be passed upwards through the control heirarchy which is why I added this method to add mouse events to all sub-controls with each option. If there is a better way to do this, feel free to add commits to this PR. (I haven't worked with windows forms much, so there definitely could be an easier solution)

**Pull request merged**